testing/e2e/agent_container_test.go

```diff
@@ -0,0 +1,238 @@
+	req.Header.Set("kbn-xsrf", "e2e-setup")
+
+	resp, err = suite.client.Do(req)
+	suite.Require().NoError(err)
+	resp.Body.Close()
+	suite.Require().Equal(http.StatusOK, resp.StatusCode, "unable to setup kibana fleet agents")
+}
+
+func (suite *AgentContainerSuite) SetupTest() {
+	// wait up to 30s  for port 8220 to be free
+	// this is a sanity check to ensure that a previous test has completely stopped before running the new test
+	portFree := false
+	for i := 0; i < 30; i++ {
+		ln, err := net.Listen("tcp", ":8220")
+		if err == nil {
+			ln.Close()
+			portFree = true
+			break
+		}
+		time.Sleep(time.Second)
+	}
+	suite.Require().True(portFree, "port 8220 must not be in use for test to start")
+}
+
+func (suite *AgentContainerSuite) TearDownTest() {
+	// stop the container when test ends
+	err := exec.Command(suite.dockerCmd, "stop", "fleet-server").Run()
+	suite.Require().NoError(err)
+}
+
+func (suite *AgentContainerSuite) TestHTTP() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, suite.dockerCmd, "run", "--rm",
+		"--name", "fleet-server",
+		"-e", "FLEET_SERVER_ENABLE=1",
+		"-e", "FLEET_SERVER_ELASTICSEARCH_HOST=http://elasticsearch:9200",
+		"-e", "FLEET_SERVER_SERVICE_TOKEN="+suite.serviceToken,
+		"-e", "FLEET_SERVER_POLICY_ID=fleet-server-policy",
+		"-e", "FLEET_SERVER_INSECURE_HTTP=true",
+		"-e", "FLEET_SERVER_HOST=0.0.0.0",
+		"-p", "8220:8220",
+		"--network", "integration_default",
+		suite.dockerImg)
+
+	err := cmd.Start()
+	suite.Require().NoError(err)
+
+	timer := time.NewTimer(time.Second)
+POLL:
+	for {
+		select {
+		case <-ctx.Done():
+			suite.Require().NoError(ctx.Err())
+		case <-timer.C:
+			// ping /api/status
+			req, err := http.NewRequestWithContext(ctx, "GET", "http://localhost:8220/api/status", nil)
+			suite.Require().NoError(err)
+
+			resp, err := suite.client.Do(req)
+			if err != nil {
+				timer.Reset(time.Second)
+				continue
+			}
+			resp.Body.Close()
+
+			// on success
+			if resp.StatusCode == http.StatusOK {
+				break POLL
+			}
+			// fail, try after a wait
+			timer.Reset(time.Second)
+		}
+	}
+	cancel()
+	cmd.Wait()
+}
+
+func (suite *AgentContainerSuite) TestWithSecretFiles() {
+	// Create a service token file in the temp test dir
+	dir := suite.T().TempDir()
+	err := os.WriteFile(filepath.Join(dir, "service-token"), []byte(suite.serviceToken), 0600)
+	suite.Require().NoError(err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, suite.dockerCmd, "run", "--rm",
+		"--name", "fleet-server",
+		"-v", suite.certPath+":/certs:ro",
+		"-v", dir+":/token:ro",
+		"-e", "FLEET_SERVER_ENABLE=1",
+		"-e", "FLEET_SERVER_CERT=/certs/fleet-server.crt",
+		"-e", "FLEET_SERVER_CERT_KEY=/certs/fleet-server.key",
+		"-e", "FLEET_SERVER_CERT_KEY_PASSPHRASE=/certs/passphrase",
+		"-e", "FLEET_SERVER_SERVICE_TOKEN_PATH=/token/service-token",
```

Comment:
Looks like I found a bug with container env vars in the agent,
Right now now the agent build from `8.8.0-ebff07ec-SNAPSHOT` fails and the logs show:
```
docker logs fleet-server
/app/apm.sh: 1: -e: not found
Policy selected for enrollment:  fleet-server-policy
Error: --fleet-server-service-token and --fleet-server-service-token-path are mutually exclusive
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.8/fleet-troubleshooting.html
Error: enrollment failed: exit status 1
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.8/fleet-troubleshooting.html
```